tests/phpunit/PaymentMethods/WC_Stripe_Express_Checkout_Element_Test.php

```diff
@@ -431,4 +431,87 @@ public function test_add_order_attribution_data() {
 		$output = ob_get_clean();
 		$this->assertStringMatchesFormat( '%aid="wc-stripe-express-checkout__order-attribution-inputs"%a', $output );
 	}
+
+	/**
+	 * Test for `is_product_page_for_ece`.
+	 *
+	 * The is_product_page_for_ece method determines whether ECE should use product
+	 * pricing vs cart pricing. It returns false when:
+	 * - Not on a product page
+	 * - On a One Page Checkout page with ECE enabled on checkout (should use cart pricing)
+	 *
+	 * @param bool $is_product               Whether we're on a product page.
+	 * @param bool $is_opc                   Whether One Page Checkout is active.
+	 * @param bool $show_ece_on_checkout     Whether ECE is enabled on checkout pages.
+	 * @param bool $expected_result          Expected return value from is_product_page_for_ece.
+	 * @return void
+	 * @dataProvider provide_test_is_product_page_for_ece
+	 */
+	public function test_is_product_page_for_ece( $is_product, $is_opc, $show_ece_on_checkout, $expected_result ) {
+		$ajax_handler = $this->getMockBuilder( WC_Stripe_Express_Checkout_Ajax_Handler::class )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$helper = $this->getMockBuilder( WC_Stripe_Express_Checkout_Helper::class )
+			->disableOriginalConstructor()
+			->setMethods( [ 'is_product', 'is_one_page_checkout', 'should_show_ece_on_checkout_page' ] )
+			->getMock();
+
+		$helper->expects( $this->any() )
+			->method( 'is_product' )
+			->willReturn( $is_product );
+
+		$helper->expects( $this->any() )
+			->method( 'is_one_page_checkout' )
+			->willReturn( $is_opc );
+
+		$helper->expects( $this->any() )
+			->method( 'should_show_ece_on_checkout_page' )
+			->willReturn( $show_ece_on_checkout );
```

Comment:
Question: can we simply drop the `expects()` statements, as they are not actually verifying any behaviour?
```suggestion
		$helper->method( 'is_product' )
			->willReturn( $is_product );

		$helper->method( 'is_one_page_checkout' )
			->willReturn( $is_opc );

		$helper->method( 'should_show_ece_on_checkout_page' )
			->willReturn( $show_ece_on_checkout );
```

Reply:
I don't think it adds any value. We have this pattern everywhere in our tests. I removed them from the files touched in this PR in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4757/commits/2270db97baf30b4119f1edcedc2d3e17c12a3a93